Add Vultr

cyberofficial opened the pull request:
This pull request adds support for several new AI models and a new provider configuration for Vultr. The main changes include the addition of five model configuration files, each describing a different model's capabilities, costs, and limits, as well as the introduction of the provider configuration for Vultr.

### New Model Configurations

* Added configuration for `DeepSeek R1 Distill Llama 70B` with detailed capabilities, cost, and limits in `deepseek-r1-distill-llama-70b.toml`.
* Added configuration for `DeepSeek R1 Distill Qwen 32B` in `deepseek-r1-distill-qwen-32b.toml`.
* Added configuration for `GPT OSS 120B` in `gpt-oss-120b.toml`.
* Added configuration for `Kimi K2 Instruct` in `kimi-k2-instruct.toml`.
* Added configuration for `Qwen2.5 Coder 32B Instruct` in `qwen2.5-coder-32b-instruct.toml`.

All models include pricing at $0.20 per million tokens for input/output.

### Provider Configuration

* Introduced the Vultr provider configuration in `provider.toml`, specifying API endpoint, npm package, environment variable, and documentation link.
* Added Vultr provider logo in `logo.svg`.

cyberofficial committed:
Add Vultr provider with 5 models, pricing, and logo

cyberofficial committed:
Update Vultr model metadata with true values

cyberofficial committed:
Update output limit to match context size

cyberofficial commented:
Converted to draft, I've put in a ticket with vultr support to get the token limits for the models, I believe they might have a different system in place, but think the numbers are correct from testing. But they are openai compatible since i've been using RooCode with it via OpenAi compat api

cyberofficial committed:
Adjust context limit in gpt-oss-120b model config

cyberofficial committed:
Adjust context limit in qwen2.5-coder model

cyberofficial committed:
Adjust context limit in kimi-k2-instruct.toml

cyberofficial commented:
Got the exact numbers from support 👍

rekram1-node commented on providers/vultr/logo.svg:
```diff
@@ -0,0 +1 @@
+<svg id="sygnet__on-white" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 42 36.09"><defs><style>.cls-1{fill:#c9f4ff;}.cls-2{fill:#51b9ff;}.cls-3{fill:#007bfc;}</style></defs><title>sygnet__on-white</title><path class="cls-1" d="M14.63,4a2.09,2.09,0,0,0-1.77-1H2.1A2.1,2.1,0,0,0,0,5.1,2.12,2.12,0,0,0,.32,6.22l2.77,4.07,13.85-3Z" transform="translate(0 -3)"/><path class="cls-2" d="M17,7.37a2.11,2. ...
```
can u make this square?

cyberofficial replied:
👍

rekram1-node commented:
open_weights is required field for the models too btw :) 

Thanks for PR!

cyberofficial committed:
Add open_weights flag to Vultr model configs

Added the open_weights property for deepseek-r1-distill-llama-70b, deepseek-r1-distill-qwen-32b, gpt-oss-120b, kimi-k2-instruct, and qwen2.5-coder-32b-instruct models. Also updated Vultr logo.svg viewBox from 36.09 to 42 to be square

cyberofficial commented:
> open_weights is required field for the models too btw :)
> 
> Thanks for PR!

I believe that should fix right?

rekram1-node commented:
lgtm thanks!
